spec/models/shelter_spec.rb

```diff
@@ -67,5 +67,41 @@
         expect(@shelter_1.pet_count).to eq(3)
       end
     end
+
+    describe "#self.order_by_recently_created" do
+      it "returns shlters ordered by most recently created first" do
+       result = Shelter.order_by_recently_created
+       expect(result).to eq([@shelter_3, @shelter_2, @shelter_1])
+      end
+    end
+
+    describe "#self.order_by_number_of_pets" do
+      it "returns shelters ordered by the number of pets they have in descending order" do
+       result = Shelter.order_by_number_of_pets
+       expect(result).to eq([@shelter_1, @shelter_3, @shelter_2])
+      end
+    end
+
+    describe "self.reverse_alphabetical" do
+      it "returns shelters ordered by name in reverse alphabetical order" do
+        result = Shelter.reverse_alphabetical
+        expect(result).to eq([@shelter_2, @shelter_3, @shelter_1])
+      end
+    end
+
+    describe "#pending_apps" do
+      it "returns shelters with pets having pending applications" do
+        johnny = Application.create!(name: 'Johnny', street_address: '1234 main st.', city: 'Westminster', state: 'CO', zipcode: '80241',  reason_for_adoption: "I love animals", status: "Pending" )
+        phylis = Application.create!(name: "Phylis", street_address: "1234 main circle", city: "Littleton", state: "CO", zipcode: "80241", reason_for_adoption: "I have a huge yard", status: "Pending")
+
+        PetApplication.create!(pet: @pet_1, application: johnny, status: "Pending")
+        PetApplication.create!(pet: @pet_2, application: phylis, status: "Approved")
+
+        result = Shelter.pending_apps
+
+        expect(result).to include(@shelter_1)
+        expect(result).not_to include(@shelter_2)
```

Comment:
Both the pets you made applications for are from @shelter_1

Can you beef this up by making applications for at least two shelters, and checking that the shelter with all approved/rejected applications doesn't show?

Reply:
Fixed - I added an approved and rejected to the second shelter instead of them both coming from shelter 1. Now it is a bit more beefy